File: Data Structures & Algorithms/validate-parentheses/submission-1.py
class Solution:
    def isValid(self, s: str) -> bool:
        if not s:
            return False
        closeToOpen = {
            ')' : '(',
            '}' : '{',
            ']' : '['
        }

        stack = []
        for n in s:
            if n in closeToOpen:
                if stack and stack[-1] == closeToOpen[n]:
                    stack.pop()
                else:
                    return False
            else:
                stack.append(n)
        return len(stack) == 0
                

File: Data Structures & Algorithms/daily-temperatures/submission-1.js
class Solution {
    /**
     * @param {number[]} temperatures
     * @return {number[]}
     */
    dailyTemperatures(temperatures) {
        let res = new Array(temperatures.length).fill(0)
        let stack = []

        for(let i = 0; i < temperatures.length; i++){
            while(stack && temperatures[i] > temperatures[stack[stack.length-1]]){
                let prevIndex = stack.pop()
                res[prevIndex] = i - prevIndex
            }
            stack.push(i)
        }
        return res
    }
}
